### PR TITLE
Resolve bug in file upload

### PR DESCRIPTION
**Pull request by chidozieononiwu**
- Update allowed APIView files

### CODE REVIEW

File: src/dotnet/APIView/APIViewWeb/Helpers/HelperAttributes.cs [hunk 1]
```diff
@@ -13,16 +13,22 @@ sealed public class UploadModelValidationAttribute : ValidationAttribute
...
             if (!String.IsNullOrEmpty(model.Language) && model.Files != null)
             {
                 var languageServices = validationContext.GetServices(typeof(LanguageService)) as IEnumerable<LanguageService>;
                 var languageService = languageServices.FirstOrDefault(s => (s as LanguageService).Name.Equals(model.Language));
-                if (!languageService.IsSupportedFile(model.Files.SingleOrDefault().FileName))
-                    return new ValidationResult($"File is invalid for the language selected. Select a file with extension {string.Join(", ", languageService.Extensions)} for language {model.Language}");
+                var fileName = model.Files.SingleOrDefault().FileName;
+                var errorMessage = $"File is invalid for the language selected. Select a file with extension {string.Join(", ", languageService.Extensions)} for language {model.Language}";
+
+                if (model.Language == "Swift" && !(fileName.EndsWith(".json") || fileName.EndsWith(".json.tgz")))
```
Comment (praveenkuttappan) [inline src/dotnet/APIView/APIViewWeb/Helpers/HelperAttributes.cs:27]: We don't support json.tgz

**Review by praveenkuttappan — APPROVED** [review]

File: src/dotnet/APIView/ClientSPA/src/app/_components/revisions-list/revisions-list.component.ts [hunk 2]
```diff
@@ -526,7 +526,7 @@ export class RevisionsListComponent implements OnInit, OnChanges {
         this.createRevisionInstruction = [
           `Run a <code>mvn package</code> build on your project, which will generate a number of build artifacts in the <code>/target</code> directory. In there, find the file ending <code>sources.jar</code>, and select it.`,
         ];
-        this.acceptedFilesForReviewUpload = ".sources.jar";
```
Comment (praveenkuttappan) [inline src/dotnet/APIView/ClientSPA/src/app/_components/revisions-list/revisions-list.component.ts:529]: We shouldn't change this. Java has .jar and .sources.jar. User must upload .sources.jar to create api review for Java.

File: src/dotnet/APIView/ClientSPA/src/app/_components/revisions-list/revisions-list.component.ts [hunk 3]
```diff
@@ -544,7 +544,7 @@ export class RevisionsListComponent implements OnInit, OnChanges {
           `Use <code>api-extractor</code> to generate a <a href="https://api-extractor.com/pages/setup/generating_docs/">docModel file</a>`,
           `Upload generated api.json file`
         ];
-        this.acceptedFilesForReviewUpload = ".api.json";
```
Comment (praveenkuttappan) [inline src/dotnet/APIView/ClientSPA/src/app/_components/revisions-list/revisions-list.component.ts:547]: This shouldn't be changes. JavaScript must upload the .api.json file which is the output of api-extractor tool they use.

**Review by praveenkuttappan — CHANGES_REQUESTED** [review]
Please revert changes for java and JavaScript file types.

File: src/dotnet/APIView/ClientSPA/src/app/_components/revisions-list/revisions-list.component.ts (hunk 3 above)
Comment (chidozieononiwu) [reply src/dotnet/APIView/ClientSPA/src/app/_components/revisions-list/revisions-list.component.ts:547]: `.api.json` is not actually an extension so it has to be `.json` or else the upload will not work. The instruction still says upload `.api.json`

**Comment by chidozieononiwu** [comment]
> Please revert changes for java and JavaScript file types.

Those were intentional. They are not errors

File: src/dotnet/APIView/ClientSPA/src/app/_components/revisions-list/revisions-list.component.ts (hunk 2 above)
Comment (chidozieononiwu) [reply src/dotnet/APIView/ClientSPA/src/app/_components/revisions-list/revisions-list.component.ts:529]: `sources.jar` is not actually an extension so it has to be `.jar` or else the upload will not work. The instruction still says upload `sources.jar`

File: src/dotnet/APIView/APIViewWeb/Helpers/HelperAttributes.cs (hunk 1 above)
Comment (chidozieononiwu) [reply src/dotnet/APIView/APIViewWeb/Helpers/HelperAttributes.cs:27]: Ok, I remove that

**Review by praveenkuttappan — APPROVED** [review]

**Pull request merged**